src/decoder.rs

```diff
@@ -739,26 +739,19 @@ fn refine_non_zeroes<R: Read>(reader: &mut R,
                               bit: i16) -> Result<u8> {
     debug_assert_eq!(coefficients.len(), 64);
 
-    let last = range.end - 1;
+    let mut last= range.start;
```

Comment:
Nitpick: Missing whitespace.

```suggestion
    let mut last = range.start;
```